src/dbhelper.py

```diff
@@ -0,0 +1,107 @@
+from PySide import QtSql
+from PySide.QtSql import QSqlQuery
+from passwords import DB_SERVER, DB_PORT, DB_LOGIN, DB_PASSWORD, DB_TABLE
+import logging
+
+# Object to wrap a database connection and provide some vaguely handy utilities.
+_logger = logging.getLogger(__name__)
+
+# Open the database connection. when the module is first touched...
+_db = QtSql.QSqlDatabase("QMYSQL")
+_db.setHostName(DB_SERVER)
+_db.setPort(DB_PORT)
+
+_db.setDatabaseName(DB_TABLE)
+_db.setUserName(DB_LOGIN)
+_db.setPassword(DB_PASSWORD)
+
+_db.setConnectOptions("MYSQL_OPT_RECONNECT=1")
+
+if not _db.open():
+    _logger.error(_db.lastError().text())
+    raise RuntimeError
+
+# A cache of prepared queries, keyed by SQL string. Allows reuse of prepared queries.
+_prepared_queries = {}
+
+def get_query_object(query_string, *query_params):
+    """
+    Get a QSqlQuery object configured for the given query string and parameters. Used by the
+    various query function to do common setup work.
+    """
+    if query_string in _prepared_queries:
+        query = _prepared_queries[query_string]
+    else:
+        query = QSqlQuery(_db)
+
+        # This ostensibly is very good for performance, and iterating resultsets backwards is a
+        # pretty strong antipattern at the best of times...
+        query.setForwardOnly(True)
+
+        query.prepare(query_string)
+        _prepared_queries[query_string] = query
+
+    # Bind the arguments to the query.
+    for i, arg in query_params:
+        query.addBindValue(arg)
+
+    return query
+
+
+def exec(query_string, *query_params):
```

Comment:
I guess we should go with `execute()` instead.

The only real potential problem though, is if someone does `from dbhelper import *`, in which case this definition will override the built-in `exec()`. But I suppose it is best practice to avoid built-in names.